### PR TITLE
OPHJOD-630: Refactor mobile navigation

### PR DESCRIPTION
**Pull request by juhaniko**
## Description

* Mobile navigation changed drastically
* Some code refactor for reusability
* Language selector highlights the current language in desktop mode too

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-630

image
image
image

### CODE REVIEW

File: src/components/LanguageButton/LanguageButton.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+interface LanguageButtonProps {
+  onLanguageClick: () => void;
+}
+
+export const LanguageButton = ({ onLanguageClick }: LanguageButtonProps) => {
+  return (
+    <button className="flex h-8 w-8 items-center justify-center rounded-full bg-bg-gray-2" onClick={onLanguageClick}>
```
Comment (juhaniko) [inline src/components/LanguageButton/LanguageButton.tsx:7]: It was _almost_ possible to use `RoundButton` here, but the icon inside the button was a bit too large ☹️

File: package.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
     "preview": "vite preview",
     "test:unit": "vitest --run",
     "test:e2e": "playwright test",
-    "prepare": "husky"
+    "prepare": "husky",
+    "update-ds": "npm update @jod/design-system"
```
Comment (sauanto) [inline package.json:19]: This is already in the readme file.

**Review by sauanto — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (juhaniko) [reply package.json:19]: I know, I was just thinking that this is more convenient. Should I remove it or update the readme to use this command?

Comment (juhaniko) [reply package.json]: Command removed

**Pull request merged**